tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/UsernamePasswordIntegrationTests.cs

```diff
@@ -76,6 +76,39 @@ public async Task AcquireTokenFromAdfsUsernamePasswordAsync()
 
         #endregion
 
+        /// <summary>
+        /// ROPC does not support MSA accounts
+        /// </summary>
+        /// <returns></returns>
+        [TestMethod]
+        public async Task ROPC_MSA_Async()
+        {
+            var labResponse = await LabUserHelper.GetMsaUserAsync().ConfigureAwait(false);
+            var user = labResponse.User;
+            labResponse.App.AppId = LabApiConstants.MSAOutlookAccountClientID;
+
+            SecureString securePassword = new NetworkCredential("", user.GetOrFetchPassword()).SecurePassword;
+
+            var msalPublicClient = PublicClientApplicationBuilder.Create(labResponse.App.AppId).WithAuthority(_authority).Build();
+
+            try
+            {
+                var result = await msalPublicClient
+                    .AcquireTokenByUsernamePassword(s_scopes, user.Upn, securePassword)
+                    .ExecuteAsync(CancellationToken.None)
+                    .ConfigureAwait(false);
+            }
+            catch (MsalServiceException ex)

```

Comment:
This is good,  but there's a helper that we tend to use called `AssertException` that pretty much wrapps the try / catch for you.

Reply:
sweet. thanks

---
In reply to: [348463937](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1511#discussion_r348463937) [](ancestors = 348463937)